### DISCUSSION TITLE
Hardware recommendation?

### DISCUSSION THREAD

**Discussion by unrly** [blakeblackshear/frigate#892] [Ask A Question]
I currently run HA in a Virtualbox VM, which just isn't cutting it at all for Frigate. I've tried multiple times to add more than 1 camera (all Wyze, no substream available) and it's just not working so I'm looking to potentially move HA to it's own server that can access the CPU (and eventually USB for a Coral device) better.

I've read the docs and saw the mention of the Atomic Pi, which I like due to the price and the low power consumption - something I'm trying to consider as I already have a 24/7 i5 server running. I'd like to stay under $75 or so and came across the Dell thin client (https://www.ebay.com/itm/Dell-N10D-Wyse-3040-Atom-x5-Z8350-Thin-Client-8GB-Flash-2GB-RAM-Thin-Computer/124414807637) that seems to be very similar to the Atomic Pi in specs - but I do wonder how this thing will perform with 5 1080p cameras or if I need to look at something a bit beefier on the CPU side.

Just wondering if anyone has any experience with this Atom CPU and if it can handle the cameras, or if I need to look at something else. Thanks

**Comment by natm** [comment]
Do you currently own a Coral? I'd certainly look at experimenting with passing that through to Virtualbox.

Also ensuring any new hardware you buy supports USB3 for the Coral at top speed.

**Reply by unrly** [reply]
I don't own one yet but I guess I might be confusing myself - does the Coral have any relation to how many cameras can be added in to Frigate? If yes, you're right in that I'd be totally overthinking this and just need to get the passthrough working to be able to handle the additional cameras.

I guess I had just assumed ffmpeg was handled by the CPU (with Coral handling AI) and that was the reason why multiple cams just wouldn't work properly but perhaps it's the detection process that's locking things up since everything is reliant on the CPU at that point.

**Reply by blakeblackshear** [reply]
It won't run as fast using VirtualBox, but if you can get the Coral passed through for detection, it will leave ffmpeg with a lot more CPU for decoding streams.

**Reply by pdecat** [reply]
FWIW, I'm using the [Coral Dual Edge TPU with a M.2 E to PCIe adapter](https://github.com/google-coral/edgetpu/issues/256#issuecomment-747942260) from a KVM/QEMU VM (Debian Buster) flawlessly:

```
<hostdev mode="subsystem" type="pci" managed="yes">
  <driver name="vfio"/>
  <source>
    <address domain="0x0000" bus="0x02" slot="0x00" function="0x0"/>
  </source>
  <alias name="hostdev2"/>
  <address type="pci" domain="0x0000" bus="0x00" slot="0x07" function="0x0"/>
</hostdev>
```

**Reply by unrly** [reply]
@pdecat That's what I was initially looking at (due to price, availability, and I'd rather plug it inside) but that setup sounds a bit over my head and apparently Virtualbox doesn't passthrough PCIE at all. I'm running Windows Server 2019 with HASS OS in Virtualbox.

**Comment by tubalainen** [comment]
I use an old Dell Core I5 (5th gen) laptop with 8 gig mem and 256 gig ssd that was donated to me.
Installed latest Debian-light/small and just installed docker-ce, intel drivers and plugged in the Coral to one of the USB 3.0 ports.
I am running 5 cams in 4x2K and 1x4K resolution at 8 fps without any issues (as of now :) ). CPU is at 50-70%
![image](https://user-images.githubusercontent.com/26795429/110479825-1baba800-80e6-11eb-9a1d-f65ffa2d5cf6.png)

**Comment by tubalainen** [comment]
AND - cannot stress this enough!

Make masks to mask away all stuff that you do NOT want to detect such as bushes, trees, sky, OSD (clock etc). That helps a HEAP with the CPU-load!
![image](https://user-images.githubusercontent.com/26795429/110480215-89f06a80-80e6-11eb-8a6f-fbe88139ea1b.png)
![image](https://user-images.githubusercontent.com/26795429/110480234-8f4db500-80e6-11eb-9f20-b9bc8d2cb131.png)

**Reply by unrly** [reply]
Thanks for the info! I also did start masking things and it seemed to help some when adding the additional cameras but not enough considering the virtualization going on. I think I'm going to try to track down a Coral and give that a shot before I go and migrate to a whole new device.
